app/Http/Controllers/ProfileController.php

```diff
@@ -135,4 +135,95 @@ public function mobileCarrier(Request $request){
             return (new ErrorResource($res))->response()->setStatusCode(400);
         }
     }
+
+    public function updateProfile(Request $request, Validate $validate){ 
+      
+       $user = $request->user();
+       $input = $request->all();
+       $validationErrors = $validate->validate($request, $this->rules->getUpdateProfileValidatonRules(),[]);
+       if( $validationErrors ){
+           return (new ErrorResource($validationErrors))->response()->setStatusCode(400);
+       }
+
+       try{           
+            $private_flags = [];
+            if(isset($input['first_name']))
+                $user->first_name =  $input['first_name']; 
+            if(isset($input['middle_name']))
```

Comment:
I probably noticed an issue here. If `middle_name` is set with some invalid characters. It'll bypass the rules validation and It will assign invalid characters in middle_name variable. You need use `nullable` or `sometimes` in the rules validation.